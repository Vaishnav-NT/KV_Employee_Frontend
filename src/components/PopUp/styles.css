.outer-div {
  height: 100vh;
  width: 100vw;
  position: fixed;
  top: 0px;
  left: 0px;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  justify-content: center;
  align-items: center;
}

.inner-div {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 400px;
  height: 209px;
  background: #ffffff 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 6px #63667a33;
  border-radius: 3px;
  opacity: 1;
}

.popup-buttons {
  display: flex;
  flex-direction: row;
}
