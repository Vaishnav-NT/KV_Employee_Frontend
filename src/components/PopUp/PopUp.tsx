import './styles.css';
import React from 'react';
import Button from '../Button/Button';

export type PopupPropsType = {
  handleConfirm: (e: any) => void;
  handleCancel: (e: any) => void;
};

const PopUp: React.FC<PopupPropsType> = ({ handleCancel, handleConfirm }) => {
  console.log(handleConfirm);

  return (
    <div className='outer-div'>
      <div className='inner-div'>
        <img src='assets/icons/close.svg' alt='close button logo' onClick={handleCancel} />
        <div className='header1'>Are you sure?</div>
        <div className='header2'>Do you really want to delete employee?</div>
        <br />
        <div className='popup-buttons'>
          <Button
            type='primary'
            label='Confirm'
            onClick={handleConfirm}
            data-testid='popup-test1'
          />
          <Button
            type='secondary'
            label='Cancel'
            onClick={handleCancel}
            data-testid='popup-test2'
          />
        </div>
      </div>
    </div>
  );
};

export default PopUp;
