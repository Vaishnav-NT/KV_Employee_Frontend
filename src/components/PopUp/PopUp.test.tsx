import PopUp from './PopUp';
import { PopupPropsType } from './PopUp';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Test if the Button component works properly', () => {
  test('To match snapshot', () => {
    const popUpProps: PopupPropsType = {
      handleCancel: () => {},
      handleConfirm: () => {}
    };

    const element = render(<PopUp {...popUpProps} />);
    expect(element).toMatchSnapshot();
  });

  test('If handleConfirm gets triggered correctly', () => {
    const handleConfirm = jest.fn();
    const popUpProps: PopupPropsType = {
      handleCancel: () => {},
      handleConfirm
    };

    render(<PopUp {...popUpProps} />);
    const element = screen.getByTestId('popup-test1');
    fireEvent.click(element);
    expect(handleConfirm).toBeCalled();
  });

  test('If handleConfirm gets triggered correctly', () => {
    const handleCancel = jest.fn();
    const popUpProps: PopupPropsType = {
      handleConfirm: () => {},
      handleCancel
    };

    render(<PopUp {...popUpProps} />);
    const element = screen.getByTestId('popup-test2');
    fireEvent.click(element);
    expect(handleCancel).toBeCalled();
  });
});
