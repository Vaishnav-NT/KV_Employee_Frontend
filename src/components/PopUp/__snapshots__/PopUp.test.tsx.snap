// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the Button component works properly To match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="outer-div"
      >
        <div
          class="inner-div"
        >
          <img
            alt="close button logo"
            src="assets/icons/close.svg"
          />
          <div
            class="header1"
          >
            Are you sure?
          </div>
          <div
            class="header2"
          >
            Do you really want to delete employee?
          </div>
          <br />
          <div
            class="popup-buttons"
          >
            <button
              class="primary"
              data-testid="button-test"
            >
              Confirm
            </button>
            <button
              class="secondary"
              data-testid="button-test"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="outer-div"
    >
      <div
        class="inner-div"
      >
        <img
          alt="close button logo"
          src="assets/icons/close.svg"
        />
        <div
          class="header1"
        >
          Are you sure?
        </div>
        <div
          class="header2"
        >
          Do you really want to delete employee?
        </div>
        <br />
        <div
          class="popup-buttons"
        >
          <button
            class="primary"
            data-testid="button-test"
          >
            Confirm
          </button>
          <button
            class="secondary"
            data-testid="button-test"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
